FinSearchUnified/Tests/Bundle/ProductNumberSearchTest.php

```diff
@@ -43,9 +47,37 @@ protected function tearDown()
         Shopware()->Container()->load('front');
         Shopware()->Container()->reset('config');
         Shopware()->Container()->load('config');
+    }
+
+    public function productNumberSearchProvider()
+    {
+        $data = '<?xml version="1.0" encoding="UTF-8"?><searchResult></searchResult>';
+        $xmlResponse = new SimpleXMLElement($data);
+
+        $query = $xmlResponse->addChild('query');
+        $query->addChild('queryString', 'queryString');
+
+        $results = $xmlResponse->addChild('results');
+        $results->addChild('count', 5);
+        $products = $xmlResponse->addChild('products');
+
+        for ($i = 1; $i <= 5; $i++) {
+            $product = $products->addChild('product');
+            $product->addAttribute('id', $i);
+        }
+
+        $filters = $xmlResponse->addChild('filters');
+        $filters->addChild('filter');
 
-        Shopware()->Session()->offsetUnset('findologicDI');
-        Shopware()->Session()->offsetUnset('isSearchPage');
+        $xml = $xmlResponse->asXML();
+
+        return [
+            'Shopware internal search, unrelated to FINDOLOGIC' => [false, true, $xml, 0],
+            'Shopware internal search' => [false, false, $xml, 0],
```

Comment:
This case has `$isUseShopSearch = false` which is not Shopware internal search or am I wrong?

Reply:
@dblaichinger actually more than this, it is the `fetchCount` variable which defines if it is an internal search or not. The `useShopSearch` only tells if FINDOLOGIC search should be active or not. Shopware sets fetchCount to false when the search is used for internal purposes.